File: src/types/github.ts
export type CheckRunInfo = {
  name: string;
  conclusion: string | null;
  status: string;
  detailsUrl: string | null;
};

export type ReviewInfo = {
  user: string;
  state: string;
  submittedAt: string | null;
};

export type PrInfo = {
  number: number;
  title: string;
  state: string;
  isDraft: boolean;
  url: string;
  checks: CheckRunInfo[];
  reviews: ReviewInfo[];
  additions: number | null;
  deletions: number | null;
  reviewDecision: string | null;
};

export type PrSummary = {
  number: number;
  title: string;
  branch: string;
  url: string;
  ciStatus: string | null;
  reviewDecision: string | null;
  repoName: string;
  author: string;
  additions: number | null;
  deletions: number | null;
  changedFiles: number | null;
  createdAt: string | null;
};

export type PrFilter = {
  author?: string | null;
  ciStatus?: string | null;
  label?: string | null;
};

export type ShepherdResult = {
  task: import('./task').TaskInfo;
  worktreePath: string;
  knowledgeRecalled: number;
};

export type QueuedRun = {
  taskPath: string;
  worktreePath: string;
};

export type QueueStatus = {
  active: string | null;
  queued: QueuedRun[];
  completed: number;
  failed: number;
};

File: src-tauri/src/models/github.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PrSummary {
    pub number: u64,
    pub title: String,
    pub branch: String,
    pub url: String,
    pub ci_status: Option<String>,
    pub review_decision: Option<String>,
    pub repo_name: String,
    pub author: String,
    pub additions: Option<u64>,
    pub deletions: Option<u64>,
    pub changed_files: Option<u64>,
    pub created_at: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone, Default)]
#[serde(rename_all = "camelCase")]
pub struct PrFilter {
    pub author: Option<String>,
    pub ci_status: Option<String>,
    pub label: Option<String>,
}
